plugins/aws_ecr/lib/samson_aws_ecr/samson_plugin.rb

```diff
@@ -0,0 +1,21 @@
+require 'aws-sdk'
+
+module SamsonAwsEcr
+  class Engine < Rails::Engine
+    mattr_accessor :ecr_client do
+      Aws::ECR::Client.new(region: ENV['AWS_REGION'] || "us-west-2")
+    end
+  end
+end
+
+Samson::Hooks.callback :before_docker_build do
+  begin
+    authorization_token = SamsonAwsEcr::Engine.ecr_client.get_authorization_token
+    authorization_data  = authorization_token.authorization_data.first
+    user, pass = Base64.decode64(authorization_data.authorization_token).split(":")
+    ENV['DOCKER_REGISTRY_USER'] = user
```

Comment:
How about you do something like store the timestamp of the last time you got a token as a `mattr_accessor`, and only get a new token if the old one is older than 8 hours, or something like that?